azerothcore/azerothcore-wotlk
# Core/Script: add hourly bell event

## Changes Proposed:

Ported hourly bell event from TC/VMangos.

Every full hour (XX:00),  clocktowers/cities in azeroth which have a bell gameobject spawn should play a hourly sound.
The sound which is played depends on the location of the spawn and the amount of tolls is determined by the current hour.
Those are the available sounds:

https://classic.wowhead.com/sound=6595
https://classic.wowhead.com/sound=6675
https://classic.wowhead.com/sound=6594
https://classic.wowhead.com/sound=6674
https://classic.wowhead.com/sound=7234
https://classic.wowhead.com/sound=7197
https://tbc.wowhead.com/sound=9154/belltollkharazhan

[update] added `tzset(); `  for `localtime_r()` in order to fix timezone/daylight saving time
https://stackoverflow.com/questions/19170721/real-time-awareness-of-timezone-change-in-localtime-vs-localtime-r

## Issues Addressed:
- 

## SOURCE:
https://github.com/vmangos/core/blob/7f38452288b3f16aaae459e0f9c9838680a5ca15/src/scripts/world/go_scripts.cpp#L345
https://github.com/TrinityCore/TrinityCore/blob/04909bada85d4770a58a5b2b6f2180f5aacb1b81/src/server/scripts/World/go_scripts.cpp#L1117

https://www.reddit.com/r/wow/comments/1ffaea/til_that_in_karazhan_a_clock_chimes_to_tell_the/

## Tests Performed:
compiles, tested ingame

## How to Test the Changes:
1. move player to a bell gameobject spawn, bell sound seems to play sound area wide
2. wait for a full hour (or if you just want to check the sound, restart the event by doing ` .event stop 73` and ` .event start 73`)
3. sound should be played.

Checklist:
  - type of sound depends on the zone/area of the spawn
  - repeat count depends on the current hour (5pm -> 5 rings)
  - event happens every a fulll hour

## Known Issues and TODO List:
- Some bell gameobject spawns are probably missing (Undercity, Scarlet Monastery ...? ) -> sniff needed
- Lighthouse sound event should be separate from hourly bell event (triggered every two mintes and sound is played once), with this PR the lighthouse sound will be played each hour like bells
- There's a missing mini bat event at brill, which is related to this game event: https://github.com/azerothcore/azerothcore-wotlk/issues/6126


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
